### ISSUE TITLE
reduce/move provider dependencies

### ISSUE THREAD

**Issue by JivanAmara** [terranodo/tegola#306]
For providers to be reused effectively, their dependencies on the larger tegola project should be
reduced.

For example, the import section of the gpkg provider includes:

- "github.com/terranodo/tegola"
- "github.com/terranodo/tegola/basic"
- "github.com/terranodo/tegola/geom"
- "github.com/terranodo/tegola/geom/encoding/wkb"
- "github.com/terranodo/tegola/internal/log"
- "github.com/terranodo/tegola/provider"

Where "github.com/terranodo/tegola" is imported only for `tegola.WebMercator`.  The geom package is a logical place for this & would eliminate this top-level dependency.

Perform similar examination of other provider dependencies .

**Comment by ARolek** [comment]
I agree. This is part of the transition to the `geom` package and purging all the top level files. After the transition we should be able to remove:

- `github.com/terranodo/tegola`
- `github.com/terranodo/tegola/basic`
